src/server/http/setup_redirect_server.js

```diff
@@ -0,0 +1,40 @@
+import { format as formatUrl } from 'url';
+import Hapi from 'hapi';
+
+// If a redirect port is specified, we need to start a server a non-ssl
+// server at this port, and redirect all requests to the ssl port.
+export default function (kbnServer, server, config) {
+  const isSslEnabled = config.get('server.ssl.enabled');
+  const portToRedirectFrom = config.get('server.ssl.redirectHttpFromPort');
+
+  // Both ssl and port to redirect from must be specified
+  if (!isSslEnabled || portToRedirectFrom === undefined) {
+    return;
+  }
+
+  const host = config.get('server.host');
+  const sslPort = config.get('server.port');
+
+  const redirectServer = new Hapi.Server();
```

Comment:
We shouldn't block on this, but I really wish we just used a native node http server for this.  I have much greater confidence in that abstraction in general than I do in hapi, both in terms of how lightweight it is, and how stable the API is.

This works though.